### PR TITLE
Change colors in the bar chart (monowiki) 

### PR DESCRIPTION
**Pull request by laudf**
In this case, I used the colormap library for set the colors.

### CODE REVIEW

**Comment by laudf** [comment]
colors
The colors visualizate like this

**Comment by javiag** [comment]
Please, do consider different colors (green, red, blue, etc) for non-ordered categories, as is the case of the namespaces shown in the picture. Consider a qualitative scale from plotly:
https://plot.ly/ipython-notebooks/color-scales/

For ordered categories a gradient between two colors makes more sense as it represents transition. In the light of the plot shown above, a transition between dark and light for the same color is not very good.  Please do consider a transition between two colors. Something like this which is also apt for colorblind people:
https://www.sciencemag.org/news/2018/08/it-s-time-use-fewer-colors-scientific-figures-some-experts-say

**Comment by Akronix** [comment]
I'm getting this error when trying to run the code:

> Traceback (most recent call last):
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
>     return self.wsgi_app(environ, start_response)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
>     response = self.handle_exception(e)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
>     reraise(exc_type, exc_value, tb)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
>     raise value
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
>     response = self.full_dispatch_request()
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
>     raise value
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/dash/dash.py", line 1073, in dispatch
>     response.set_data(self.callback_map[output]['callback'](*args))
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/dash/dash.py", line 969, in add_context
>     output_value = func(*args, **kwargs)
>   File "/home/akronix/workspace/wikis_research/WikiChron/wikichron/dash/apps/monowiki/main.py", line 535, in update_graphs
>     new_graphs = generate_graphs(data, metrics, wikis, relative_time)
>   File "/home/akronix/workspace/wikis_research/WikiChron/wikichron/dash/apps/monowiki/main.py", line 86, in generate_graphs
>     colors_rgb = colors_bar_chart()
>   File "/home/akronix/workspace/wikis_research/WikiChron/wikichron/dash/apps/monowiki/main.py", line 43, in colors_bar_chart
>     c.plot_colormap('sequentials')
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/colormap/colors.py", line 958, in plot_colormap
>     from pylab import subplots
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/pylab.py", line 1, in <module>
>     from matplotlib.pylab import *
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/matplotlib/pylab.py", line 245, in <module>
>     from matplotlib import cbook, mlab, pyplot as plt
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/matplotlib/pyplot.py", line 2372, in <module>
>     switch_backend(rcParams["backend"])
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/matplotlib/pyplot.py", line 207, in switch_backend
>     backend_mod = importlib.import_module(backend_name)
>   File "/home/akronix/.pyenv/versions/3.6.7/lib/python3.6/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/matplotlib/backends/backend_tkagg.py", line 1, in <module>
>     from . import _backend_tk
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/matplotlib/backends/_backend_tk.py", line 5, in <module>
>     import tkinter as Tk
>   File "/home/akronix/.pyenv/versions/3.6.7/lib/python3.6/tkinter/__init__.py", line 36, in <module>
>     import _tkinter # If this fails your Python may not be configured for Tk
> ModuleNotFoundError: No module named '_tkinter'

Do you know how to fix it?

**Comment by laudf** [comment]
colors
colors1
@javiag better?

**Comment by laudf** [comment]
@Akronix try to remove the line 43 and run it, please. I don´t get an error because python for windows has installed "tkinter" by default.

**Comment by javiag** [comment]
Thanks @laudf ! 

It's definitely getting there, but I think that in the first case the shades of green are too similar to clearly see them.  Sorry about that. This is trial an error.  Perhpas, with transition between three colors...

Regarding the second plot, the grey color is a bit ugly, but other than that it could work.

**Comment by Akronix** [comment]
Now, when selecting http://localhost:5002/monowiki/app/?wikis=cocktails.fandom.com&metrics=40_users_edits_number_of_edits, I'm getting this error:

> Traceback (most recent call last):
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
>     return self.wsgi_app(environ, start_response)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
>     response = self.handle_exception(e)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
>     reraise(exc_type, exc_value, tb)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
>     raise value
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
>     response = self.full_dispatch_request()
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
>     raise value
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/dash/dash.py", line 1073, in dispatch
>     response.set_data(self.callback_map[output]['callback'](*args))
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/dash/dash.py", line 969, in add_context
>     output_value = func(*args, **kwargs)
>   File "/home/akronix/workspace/wikis_research/WikiChron/wikichron/dash/apps/monowiki/main.py", line 581, in update_graphs
>     new_graphs = generate_graphs(data, metrics, wikis, relative_time)
>   File "/home/akronix/workspace/wikis_research/WikiChron/wikichron/dash/apps/monowiki/main.py", line 90, in generate_graphs
>     mycmap = c.cmap('YlGnBu_r')
>   File "/home/akronix/.pyenv/versions/3.6.7/envs/wikichron/lib/python3.6/site-packages/colormap/colors.py", line 860, in cmap
>     index = np.linspace(0, 1, len(colors['red']))
> TypeError: string indices must be integers

**Comment by laudf** [comment]
@javiag 
colors
What do you think? Is it better?
colors1
And in this case, i think the difference  between colors looks better like this. What do you think?

**Comment by javiag** [comment]
It looks definitely much better. I think we can consider the second one closed.

Regarding the first one, "Others" definitely should be on top of the bar, while Articles in the bottom. The both talk pages should be together, since they are related. The order should be: articles, article talk, user talk, user, templates, others. Regarding the colors, change the pink. Or better, use plotly "qualitative" set for 7 colors and discard one.

**Comment by Akronix** [comment]
I keep getting the error posted in [this comment](https://github.com/Grasia/WikiChron/pull/84#issuecomment-505839396) with the same selection: http://localhost:5002/monowiki/app/?wikis=cocktails.fandom.com&metrics=40_users_edits_number_of_edits but with other selections as well: http://localhost:5002/monowiki/app/?wikis=cocktails.fandom.com&metrics=41_users_first_edit.

Please read @javiag's observations and take care of them.

File: wikichron/dash/apps/monowiki/main.py [hunk 1]
```diff
@@ -79,6 +79,7 @@ def generate_graphs(metrics, wikis, relative_time):
 
     for metric_idx in range(len(metrics)):
         graphs_list.append([])
+<<<<<<< HEAD
```
Comment (Akronix) [inline wikichron/dash/apps/monowiki/main.py]: There git comments should not be in source code.

File: wikichron/dash/apps/monowiki/main.py [hunk 2]
```diff
@@ -200,6 +201,12 @@ def generate_graphs(metrics, wikis, relative_time):
                                     name=metric_data.name
                                     )
 
+=======
```
Comment (Akronix) [inline wikichron/dash/apps/monowiki/main.py]: There should not be git comments in source code.

**Comment by Akronix** [comment]
Closing because code is out of date.

**Pull request closed without merging**